internal/grpc/xds.go

```diff
@@ -40,10 +40,10 @@ func (xh *xdsHandler) fetch(req *v2.DiscoveryRequest) (*v2.DiscoveryResponse, er
 	}
 	resources, err := toAny(r, toFilter(req.ResourceNames))
 	return &v2.DiscoveryResponse{
-		VersionInfo: "0",
+		VersionInfo: req.VersionInfo,
```

Comment:
Does this fix the issue (I would love it if it does)??  I'll go test the branch ;) . Looking at the xDS protocol diagram, the management server is suppose to control version and nonce values, passing them back as part of DiscoveryResponse to be used in the NACK/ACK DiscoveryRequest.
 - https://github.com/envoyproxy/data-plane-api/blob/master/XDS_PROTOCOL.md

Reply:
At the moment we always send version_info="0", which is not correct, it should _change_. Looking through the code on the envoy side there is no requirement for version_info to increment, I've not found any place where two versions are sorted, they are only compared for equality.

I'm not sure this fix is correct, in the initial discovery request version info will be "", so rather than replying with "0", we'd reply with "" which has some unexpected consequences as "" is treated as a marker for some conditions.

This is also why the tests pass with the versioninfo and nonce fields removed in all the `internal/e2e` test cases. I don't think this is correct.

It's easy enough to get the right version info, it's in `last`. We can use the string form of `last` for VersionInfo.

For the nonce, this is only used in the outer loop of `GrpcMuxImpl::onDiscoveryResponse` on the Envoy side. Its never compared to a previous nonce, or sorted for newer or olderness, despite what the xDS protocol document says. This is sort of backed up by the observation that if these were serial numbers, they would be sent on the wire as numbers, not opaque strings.

I think for the moment we can use the string form of last here, at least it will change on every response.